vispy/visuals/colorbar.py

```diff
@@ -0,0 +1,398 @@
+            text_y = y
+            self._label.pos = text_x, text_y
+            self._label.rotation = -90
+
+            # TODO, HACK: See comment about ticks on "horizontal" conditional
+            self._ticks[0].pos = text_x, y + halfh
+            self._ticks[1].pos = text_x, y - halfh
+            self._ticks[0].rotation = self.ticks[1].rotation = -90
+
+        elif self._orientation == "left":
+            text_x = x - _TEXT_PADDING_FACTOR * (halfw + self.border_width)
+            text_y = y
+            self._label.pos = text_x, text_y
+            self._label.rotation = -270
+
+            self._ticks[0].pos = text_x, y + halfh
+            self._ticks[1].pos = text_x, y - halfh
+            self._ticks[0].rotation = self.ticks[1].rotation = -270
+
+        else:
+            # raise an error since the orientation is now what was
+            # expected
+            raise ColorBarVisual._get_orientation_error(self._orientation)
+
+        # Set up the attributes that the shaders require
+        vertices = np.array([[x - halfw, y - halfh],
+                            [x + halfw, y - halfh],
+                            [x + halfw, y + halfh],
+                             # tri 2
+                            [x - halfw, y - halfh],
+                            [x + halfw, y + halfh],
+                            [x - halfw, y + halfh]],
+                            dtype=np.float32)
+
+        self._program['a_position'] = vertices.astype(np.float32)
+        self._border_program['a_position'] = vertices.astype(np.float32)
+
+        self._border_program.vert['border_width'] = self._border_width
+        self._border_program.frag['border_color'] = self._border_color.rgba
+
+    @staticmethod
+    def _get_orientation_error(orientation):
+        return ValueError("orientation must"
+                          " be 'horizontal' or 'vertical', "
+                          "not '%s'" % (orientation, ))
+
+    @staticmethod
+    def _get_anchors(orientation):
+        if orientation == "top":
+            return "center", "bottom"
+        elif orientation == "bottom":
+            return "center", "top"
+        elif orientation == "left":
+            return "center", "top"
+        else:  # orientation == "right"
+            return "center", "top"
+
+    @property
+    def cmap(self):
+        return self._cmap
+
+    @cmap.setter
+    def cmap(self, cmap):
+        self._cmap = get_colormap(cmap)
+        self._program.frag['color_transform'] = Function(self._cmap.glsl_map)
+
+    @property
+    def clim(self):
+        return self._clim
+
+    @clim.setter
+    def clim(self, clim):
+        self._clim = clim
+        # new TextVisuals need to be created
+        self._update()
+
+    @property
+    def label(self):
+        return self._label
+
+    @label.setter
+    def label(self, label):
+        self._label = label
+        # position, styling has to be applied
+        self._update()
+
+    @property
+    def ticks(self):
```

Comment:
Same problem as that of ```label```. I'm giving the user direct access, which I'm not too fond of